### ISSUE TITLE
Crashing when opening.

### ISSUE THREAD

**Issue by maumagro** [OpenLoco/OpenLoco#2401]
**Version information:**
 - Platform: Windows 11 Pro 
 - Version: 23H2 (Build 22631.3296)
 - Build/commit: f0863a0c

**Describe the bug**
When opening OpenLoco from the Launcher (also with zipped version), I get a crash dump. Tried changing the display settings on openloco.yml as I read on an article, still same problem. 

**To Reproduce**
Steps to reproduce the behaviour:
Just opening the game from launcher or openloco.exe

**Expected behaviour**
Game opening and having fun.

**Screenshots**
![image](https://github.com/OpenLoco/OpenLoco/assets/61612545/e117d0b3-2ce8-4d94-acc1-4e7bb16a87cc)

**Additional context**
Can't attach crash dump, weighs 61 MB.

**Comment by maumagro** [comment]
[1c7e7732-0b87-48c5-8189-a03c49ab6b6b(OpenLoco, v24.03 (f0863a0c on master)).zip](https://github.com/OpenLoco/OpenLoco/files/14876266/1c7e7732-0b87-48c5-8189-a03c49ab6b6b.OpenLoco.v24.03.f0863a0c.on.master.zip)

Why didn't I think of zipping it? Here you go. Thank you!

**Comment by duncanspumpkin** [comment]
This is crashing in an opengl init function which we don't have much to do with. All we do is tell SDL to get it setup, tell it the dimensions of the window and make a suggestion of using opengl. Could you provide your openloco.yml file? Maybe the width and height are something odd.

**Comment by maumagro** [comment]
Here's the openloco.yml

---------------------

display:
  mode: fullscreen
  window_resolution:
    width: 800
    height: 600
  fullscreen_resolution:
    width: 1920
    height: 1080
audio:
  play_title_music: true
network:
  enabled: false
loco_install_path: C:\Program Files (x86)\GOG Galaxy\Games\Locomotion
last_save_path: ""
last_landscape_path: ""
language: en-GB
preferredCurrency:
  flags: 130
  name: CURRDOLL
  checksum: 0
usePreferredCurrencyForNewGames: false
usePreferredCurrencyAlways: false
scale_factor: 1
showFPS: false
uncapFPS: false
allow_multiple_instances: false
cashPopupRendering: true
edgeScrolling: true
edgeScrollingSpeed: 12
zoom_to_cursor: true
autosave_amount: 12
autosave_frequency: 1
breakdowns_disabled: false
buildLockedVehicles: false
cheats_menu_enabled: false
companyAIDisabled: false
disableVehicleLoadPenaltyCheat: false
displayLockedVehicles: false
invertRightMouseViewPan: false
townGrowthDisabled: false
trainsReverseAtSignals: true
disableStationSizeLimit: false
preferredOwnerName: ""
usePreferredOwnerName: false
preferredOwnerFace:
  flags: 0
  name: "\x00\x00\x00\x00\x00\x00\x00\x00"
  checksum: 0
usePreferredOwnerFace: false
shortcuts:
  closeTopmostWindow: Backspace
  closeAllFloatingWindows: Left Shift+Backspace
  cancelConstructionMode: Escape
  pauseUnpauseGame: Pause
  zoomViewOut: PageUp
  zoomViewIn: PageDown
  rotateView: Return
  rotateConstructionObject: Z
  toggleUndergroundView: 1
  toggleSeeThroughTracks: 2
  toggleSeeThroughRoads: 3
  toggleSeeThroughTrees: 4
  toggleSeeThroughBuildings: 5
  toggleSeeThroughScenery: 6
  toggleHeightMarksOnLand: 7
  toggleHeightMarksOnTracks: 8
  toggleDirArrowsOnTracks: 9
  adjustLand: L
  adjustWater: W
  plantTrees: P
  bulldozeArea: X
  buildTracks: T
  buildRoads: R
  buildAirports: A
  buildShipPorts: D
  buildNewVehicles: N
  showVehiclesList: V
  showStationsList: S
  showTownsList: U
  showIndustriesList: I
  showMap: M
  showCompaniesList: C
  showCompanyInformation: Q
  showFinances: F
  showAnnouncementsList: Tab
  makeScreenshot: Left Ctrl+S
  toggleLastAnnouncement: Space
  sendMessage: F1
  constructionPreviousTab: ""
  constructionNextTab: ""
  constructionPreviousTrackPiece: ""
  constructionNextTrackPiece: ""
  constructionPreviousSlope: ""
  constructionNextSlope: ""
  constructionBuildAtCurrentPos: ""
  constructionRemoveAtCurrentPos: ""
  constructionSelectPosition: ""
  gameSpeedNormal: ""
  gameSpeedFastForward: ""
  gameSpeedExtraFastForward: ""

**Comment by duncanspumpkin** [comment]
Hmm that's annoying. Nothing unique or odd there. I'm out of suggestions. Update graphics card drivers? Do you play other games do they have issues? It looks like an sdl issue which is used by tons of games.

**Comment by maumagro** [comment]
I play tons of games, most regular are AoE II Definitive, GTA V, Ark, Cities Skylines, Transport Fever, Portal 2, FSX. I hope that helps.

**Comment by AaronVanGeffen** [comment]
I think we should update our SDL dependencies, as they're pinned to one of last year's versions of SDL. Not sure if it will help your case, but it's worth a try.

Could you try a build from #2407 to see if that fixes the issue for you? Once CI has finished, they should appear at the bottom of [this page](https://github.com/OpenLoco/OpenLoco/actions/runs/8587101840?pr=2407).

**Comment by AaronVanGeffen** [comment]
We released v24.04 moments ago, which uses an updated version of the SDL library. Could you try the new version, and see if that solves your problem?

**Comment by LeftofZen** [comment]
That's unfortunate then, I'm not sure how we'll debug this one then

**Comment by LeftofZen** [comment]
Looks like you have an ATI graphics card? Can you download the latest drivers for your card? What card specifically is it?

**Comment by maumagro** [comment]
I've 2 graphic cards. GTX 960 2GB and AMD Radeon R7 IGP. Tried changing both on graphic settings on windows, same thing.

![image](https://github.com/OpenLoco/OpenLoco/assets/61612545/7f5ff5e0-a42a-4dd4-b354-8c25be58cffa)

**Comment by LeftofZen** [comment]
yep ok, the 960 is definitely the way to go as while loco doesn't actually use the gpu for any hardward rendering, the IGP probably has some funky stuff going on. the 960 is quite old though - what driver version does it say you have in geforce experience?

**Comment by maumagro** [comment]
Updated 4 days ago.

Yes, the GPU is quite old, but I don't have any problems with any other game, everything works like a charm.

![image](https://github.com/OpenLoco/OpenLoco/assets/61612545/dbb1a12f-71d2-437c-a8ff-e124e00024f3)

**Comment by LeftofZen** [comment]
hmm yeah i don't see any reason why it wouldn't work then. although i'm still puzzled why the amd/ati drivers are in the minidump call stack, as that implies the integrated gpu was used during the crash

**Comment by LeftofZen** [comment]
I recall on my old laptop, there was also an nvidia setting for forcing the GPU. I really don't trust Windows to enforce it, so could you try this option as well

https://superuser.com/questions/1815471/force-windows-laptop-to-always-use-my-dedicated-gpu-on-power-and-integrated-gpu

**Comment by LeftofZen** [comment]
in fact can you set it to `High-performance NVIDIA processor` and then run `dxdiag command?

![image](https://github.com/OpenLoco/OpenLoco/assets/7483209/4537a110-37af-4ea3-96f5-010dadd2edf4)

and then printscreen the output for display1, like this

![image](https://github.com/OpenLoco/OpenLoco/assets/7483209/35e8b857-35ac-4c42-a7ba-5c86f85af7c3)

**Comment by shusaura85** [comment]
Issue is stale. Can probably be closed
